framework/src/meshgenerators/CutMeshByLevelSetGeneratorBase.C

```diff
@@ -0,0 +1,509 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CutMeshByLevelSetGeneratorBase.h"
+#include "MooseMeshElementConversionUtils.h"
+#include "MooseMeshUtils.h"
+
+#include "libmesh/elem.h"
+#include "libmesh/boundary_info.h"
+#include "libmesh/mesh_base.h"
+#include "libmesh/parallel.h"
+#include "libmesh/parallel_algebra.h"
+#include "libmesh/cell_tet4.h"
+
+// C++ includes
+#include <cmath>
+
+InputParameters
+CutMeshByLevelSetGeneratorBase::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+  params += FunctionParserUtils<false>::validParams();
+
+  params.addRequiredParam<MeshGeneratorName>("input", "The input mesh that needs to be trimmed.");
+
+  params.addParam<boundary_id_type>("cut_face_id",
+                                    "The boundary id of the face generated by the cut. An "
+                                    "id will be automatically assigned if not provided.");
+  params.addParam<BoundaryName>("cut_face_name",
+                                "The boundary name of the face generated by the cut.");
+
+  params.addClassDescription(
+      "This CutMeshByLevelSetGeneratorBase object is designed to be the base class of mesh "
+      "generator that cuts a 3D mesh based on a analytic level set function. The level set "
+      "function could be provided explicitly or indirectly.");
+
+  return params;
+}
+
+CutMeshByLevelSetGeneratorBase::CutMeshByLevelSetGeneratorBase(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    FunctionParserUtils<false>(parameters),
+    _input_name(getParam<MeshGeneratorName>("input")),
+    _cut_face_name(isParamValid("cut_face_name") ? getParam<BoundaryName>("cut_face_name")
+                                                 : BoundaryName()),
```

Comment:
let's have a default instead of this

Reply:
a default name may cause conflicting issue if users cut a mesh more than once.

Reply:
a default of "" for the parameter, which then does not get mapped to the id